third-party/rspamd/src/main/java/org/apache/james/rspamd/task/GetMailboxMessagesService.java

```diff
@@ -90,9 +90,7 @@ private Flux<MessageResult> getMailboxMessagesOfAUser(Username username, String
             .doOnNext(mailboxMessageMetaData -> context.incrementSpamMessageCount())
             .filter(mailboxMessageMetaData -> afterDate.map(date -> mailboxMessageMetaData.getInternalDate().after(date)).orElse(true))
             .filter(message -> randomBooleanWithProbability(runningOptions))
-            .map(Message::getMessageId)
-            .collectList()
-            .flatMapMany(messageIds -> messageIdManager.getMessagesReactive(messageIds, FetchGroup.FULL_CONTENT, mailboxSession))
+            .flatMap(message -> messageIdManager.getMessagesReactive(List.of(message.getMessageId()), FetchGroup.FULL_CONTENT, mailboxSession))
```

Comment:
We likely can use window operator to still read msgs by batch without reading them all

Reply:
"window" is expensive, should we still consider in this case?

Reply:
No window by item count is cheap, but likely not necessary perf wise. Let's go ahead imo